### PR TITLE
Added more test on tool library

### PR DESCRIPTION
**Pull request by sima-zhu**

### CODE REVIEW

File: tools/code-owners-parser/Azure.Sdk.Tools.RetrieveCodeOwners.Tests/MainTests.cs [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+            new Tuple<string, bool, List<string>>("sdk", false, new List<string> { "person1", "person2" }),
+            new Tuple<string, bool, List<string>>("/sdk", false, new List<string> { "person1", "person2" }),
+            new Tuple<string, bool, List<string>>("sdk/noPath", false, new List<string> { "person1", "person2"}),
+            new Tuple<string, bool, List<string>>("/sdk/azconfig", false, new List<string> { "person3", "person4"}),
+            new Tuple<string, bool, List<string>>("/sdk/azconfig/package", false, new List<string> { "person3", "person4"}),
+            new Tuple<string, bool, List<string>>("/sdk/testUser/", true, new List<string> { "azure-sdk" }),
+            new Tuple<string, bool, List<string>>("/sd", true, new List<string>()),
+        };
+
+        [Test]
```
Comment (weshaggard) [inline tools/code-owners-parser/Azure.Sdk.Tools.RetrieveCodeOwners.Tests/MainTests.cs:23]: Instead of having only a single test you should consider testcase https://docs.nunit.org/articles/nunit/writing-tests/attributes/testcase.html so that we have better reporting of which case is actually failing.

File: tools/code-owners-parser/Azure.Sdk.Tools.RetrieveCodeOwners.Tests/MainTests.cs [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        public void testOnError()
+        {
+            Assert.AreEqual(1, Program.Main("PathNotExist", "sdk"));
+        }
+
+        private void testExpectResult(List<string> expectReturn, string output)
+        {
+            if (output == null)
+            {
+
```
Comment (weshaggard) [inline tools/code-owners-parser/Azure.Sdk.Tools.RetrieveCodeOwners.Tests/MainTests.cs:48]: Did you want to do something here?

File: tools/code-owners-parser/ci.yml [hunk 3]
```diff
@@ -29,9 +29,13 @@ extends:
...
     TestPostSteps:
       - task: PowerShell@2
-        displayName: Test on code owner tool output
+        displayName: Test on code owner script
         inputs:
           pwsh: true
           filePath: eng/common/scripts/get-codeowners.ps1
           arguments: >
             -Test
+      - pwsh: |
```
Comment (weshaggard) [inline tools/code-owners-parser/ci.yml:38]: You shouldn't need to add this as we already do it in the common template.

File: tools/code-owners-parser/Azure.Sdk.Tools.RetrieveCodeOwners.Tests/MainTests.cs (hunk 1 above)
Comment (sima-zhu) [reply tools/code-owners-parser/Azure.Sdk.Tools.RetrieveCodeOwners.Tests/MainTests.cs:23]: I am just wondering if we have something like parameterized tests. Thanks for pointing this out.

File: tools/code-owners-parser/Azure.Sdk.Tools.RetrieveCodeOwners.Tests/MainTests.cs (hunk 2 above)
Comment (sima-zhu) [reply tools/code-owners-parser/Azure.Sdk.Tools.RetrieveCodeOwners.Tests/MainTests.cs:48]: No. Will remove

File: tools/code-owners-parser/ci.yml (hunk 3 above)
Comment (sima-zhu) [reply tools/code-owners-parser/ci.yml:38]: The common one actually test in Azure.Sdk.Tools.RetrieveCodeOwners
This one tests under Azure.Sdk.Tools.RetrieveCodeOwners.Tests

Comment (weshaggard) [reply tools/code-owners-parser/ci.yml:38]: Have a look at https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1213179&view=logs&j=f1c271e7-ab66-52b3-53cf-7a71849a1672&t=c5205d3b-39fe-5155-0a53-cf094d5edf2e

It calls dotnet test on all the projects so this explicit step shouldn't be needed.

Comment (weshaggard) [reply tools/code-owners-parser/ci.yml:38]: Also it doesn't look like your tests are running correctly. Please add a failing test to make sure things are getting discovered and ran correctly.

File: tools/code-owners-parser/Azure.Sdk.Tools.RetrieveCodeOwners.Tests/MainTests.cs [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+        private const string codeOwnerFilePath = "CODEOWNERS";
+
+        [TestCase("sdk", false, new[] { "person1", "person2" })]
+        [TestCase("/sdk", false, new[] { "person1", "person2" })]
+        [TestCase("sdk/noPath", false, new[] { "person1", "person2" })]
+        [TestCase("/sdk/azconfig", false, new[] { "person3", "person4" })]
+        [TestCase("/sdk/azconfig/package", false, new[] { "person3", "person4" })]
+        [TestCase("/sdk/testUser/", true, new[] { "azure-sdk" })]
+        [TestCase("/sd", true, new string[0])]
+        public void testOnNormalOuput(string targetDirectory, bool includeUserAliasesOnly, List<string> expectedReturn)
```
Comment (weshaggard) [inline tools/code-owners-parser/Azure.Sdk.Tools.RetrieveCodeOwners.Tests/MainTests.cs:19]:
```suggestion
        public void TestOnNormalOuput(string targetDirectory, bool includeUserAliasesOnly, List<string> expectedReturn)
```

File: tools/code-owners-parser/Azure.Sdk.Tools.RetrieveCodeOwners.Tests/MainTests.cs [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+            {
+                Program.Main(codeOwnerFilePath, targetDirectory, includeUserAliasesOnly);
+                var output = consoleOutput.GetOuput();
+                testExpectResult(expectedReturn, output);
+                consoleOutput.Dispose();
+            }
+        }
+
+        [Test]
+        public void testOnError()
```
Comment (weshaggard) [inline tools/code-owners-parser/Azure.Sdk.Tools.RetrieveCodeOwners.Tests/MainTests.cs:32]:
```suggestion
        public void TestOnError()
```

File: tools/code-owners-parser/Azure.Sdk.Tools.RetrieveCodeOwners.Tests/MainTests.cs [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+        {
+            Assert.AreEqual(1, Program.Main("PathNotExist", "sdk"));
+        }
+
+        private void testExpectResult(List<string> expectReturn, string output)
```
Comment (weshaggard) [inline tools/code-owners-parser/Azure.Sdk.Tools.RetrieveCodeOwners.Tests/MainTests.cs:37]:
```suggestion
        private void TestExpectResult(List<string> expectReturn, string output)
```

Methods in .NET are supposed to start with capital letters by convention.

File: tools/code-owners-parser/CodeOwnersParser/Azure.Sdk.Tools.CodeOwnersParser.csproj [hunk 7]
```diff
@@ -3,6 +3,7 @@
   <PropertyGroup>
     <IsPackable>false</IsPackable>
     <TargetFramework>netstandard2.0</TargetFramework>
+    <Platforms>AnyCPU;x64</Platforms>
```
Comment (weshaggard) [inline tools/code-owners-parser/CodeOwnersParser/Azure.Sdk.Tools.CodeOwnersParser.csproj:6]: Why is this needed?

File: tools/code-owners-parser/Azure.Sdk.Tools.RetrieveCodeOwners.Tests/Azure.Sdk.Tools.RetrieveCodeOwners.Tests.csproj [hunk 8]
```diff
@@ -0,0 +1,25 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>net5.0</TargetFramework>
+
+    <IsPackable>false</IsPackable>
+
+    <PlatformTarget>x64</PlatformTarget>
```
Comment (weshaggard) [inline tools/code-owners-parser/Azure.Sdk.Tools.RetrieveCodeOwners.Tests/Azure.Sdk.Tools.RetrieveCodeOwners.Tests.csproj:8]: Why is this needed?

File: tools/code-owners-parser/ci.yml (hunk 3 above)
Comment (sima-zhu) [reply tools/code-owners-parser/ci.yml]: I missed one of the dependencies when use Testcase. It should work for latest changes

File: tools/code-owners-parser/Azure.Sdk.Tools.RetrieveCodeOwners.Tests/Azure.Sdk.Tools.RetrieveCodeOwners.Tests.csproj (hunk 8 above)
Comment (sima-zhu) [reply tools/code-owners-parser/Azure.Sdk.Tools.RetrieveCodeOwners.Tests/Azure.Sdk.Tools.RetrieveCodeOwners.Tests.csproj]: Removed

File: tools/code-owners-parser/CodeOwnersParser/Azure.Sdk.Tools.CodeOwnersParser.csproj (hunk 7 above)
Comment (sima-zhu) [reply tools/code-owners-parser/CodeOwnersParser/Azure.Sdk.Tools.CodeOwnersParser.csproj]: Removed

File: tools/code-owners-parser/CodeOwnersParser/FileHelpers.cs [hunk 9]
```diff
@@ -8,24 +8,22 @@ public static class FileHelpers
     {
         public static string GetFileContents(string fileOrUri)
         {
-            if (File.Exists(fileOrUri))
-            {
-                return File.ReadAllText(fileOrUri);
-            }
-
+            string fullPath = (new DirectoryInfo(fileOrUri)).FullName;
```
Comment (weshaggard) [inline tools/code-owners-parser/CodeOwnersParser/FileHelpers.cs:11]: Why are you creating a DirectoryInfo here?

File: tools/code-owners-parser/CodeOwnersParser/FileHelpers.cs [hunk 10]
```diff
@@ -8,24 +8,22 @@ public static class FileHelpers
...
-            // try to download it.
-            using (HttpClient client = new HttpClient())
+            if (File.Exists(fullPath))
             {
-                HttpResponseMessage response = client.GetAsync(fileOrUri).ConfigureAwait(false).GetAwaiter().GetResult();
-                return response.Content.ReadAsStringAsync().ConfigureAwait(false).GetAwaiter().GetResult();
+                return File.ReadAllText(fullPath);
             }
+
+            throw new ArgumentException(string.Format("The path provided is neither local path nor https link. Please check your path: {0}", fileOrUri));
```
Comment (weshaggard) [inline tools/code-owners-parser/CodeOwnersParser/FileHelpers.cs:26]: tip:
```suggestion
            throw new ArgumentException($"The path provided is neither local path nor https link. Please check your path: {fileOrUri}");
```

File: tools/code-owners-parser/CodeOwnersParser/FileHelpers.cs (hunk 9 above)
Comment (sima-zhu) [reply tools/code-owners-parser/CodeOwnersParser/FileHelpers.cs]: Changed as suggested. Use `Path.GetFullPath`

File: tools/code-owners-parser/CodeOwnersParser/FileHelpers.cs [hunk 11]
```diff
@@ -8,24 +8,23 @@ public static class FileHelpers
...
-            using (HttpClient client = new HttpClient())
+            string fullPath = Path.GetFullPath(fileOrUri);
+            if (File.Exists(fullPath))
             {
-                HttpResponseMessage response = client.GetAsync(fileOrUri).ConfigureAwait(false).GetAwaiter().GetResult();
-                return response.Content.ReadAsStringAsync().ConfigureAwait(false).GetAwaiter().GetResult();
+                return File.ReadAllText(fullPath);
             }
+
+            throw new ArgumentException($"The path provided is neither local path nor https link. Please check your path: {fileOrUri}");
```
Comment (weshaggard) [inline tools/code-owners-parser/CodeOwnersParser/FileHelpers.cs:27]:
```suggestion
            throw new ArgumentException($"The path provided is neither local path nor https link. Please check your path: {fullPath}");
```

**Review by weshaggard — APPROVED** [review]
